solr/solrj/src/java/org/apache/solr/common/util/CompressionUtil.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.common.util;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.zip.DataFormatException;
+import java.util.zip.Deflater;
+import java.util.zip.Inflater;
+
+/**
+ * Utility class for dealing with LZ4 compressed data, such as state.json in Zookeeper if configured
```

Comment:
I actually wish it was LZ4 but you're using ZLIB.  I think the methods on this class should indicate the compression used.  I imagine we might want to actually use LZ4 and then we'd have a naming issue.

Reply:
We actually originally started down the path of LZ4, but a number of different tools/languages we use have built in support for ZLIB compression vs having to use a third party library or build additional support for LZ4. I can do some refactoring here to make it open to additional compression methods. Out of curiosity, what would push you towards LZ4 instead?

Reply:
I recall the compress/decompress speed was better even if it didn't shrink the size as much as ZLIB, which seemed to be a better trade-off for this use-case.  But I totally understand wanting to prefer convenience of ease of access to ZLIB in tooling.